provider/cloudflare/cloudflare.go

```diff
@@ -140,25 +140,71 @@ type RecordParamsTypes interface {
 }
 
 // getUpdateDNSRecordParam is a function that returns the appropriate Record Param based on the cloudFlareChange passed in
-func getUpdateDNSRecordParam(cfc cloudFlareChange) cloudflare.UpdateDNSRecordParams {
-	return cloudflare.UpdateDNSRecordParams{
+func getUpdateDNSRecordParam(cfc cloudFlareChange) (cloudflare.UpdateDNSRecordParams, error) {
+	recordParam := cloudflare.UpdateDNSRecordParams{
 		Name:    cfc.ResourceRecord.Name,
 		TTL:     cfc.ResourceRecord.TTL,
 		Proxied: cfc.ResourceRecord.Proxied,
 		Type:    cfc.ResourceRecord.Type,
-		Content: cfc.ResourceRecord.Content,
 	}
+
+	if cfc.ResourceRecord.Type == "SRV" {
+		recordParam.Data = cfc.ResourceRecord.Data
+	} else {
+		recordParam.Content = cfc.ResourceRecord.Content
+	}
+
+	return recordParam, nil
 }
 
 // getCreateDNSRecordParam is a function that returns the appropriate Record Param based on the cloudFlareChange passed in
-func getCreateDNSRecordParam(cfc cloudFlareChange) cloudflare.CreateDNSRecordParams {
-	return cloudflare.CreateDNSRecordParams{
+func getCreateDNSRecordParam(cfc cloudFlareChange) (cloudflare.CreateDNSRecordParams, error) {
+	recordParam := cloudflare.CreateDNSRecordParams{
 		Name:    cfc.ResourceRecord.Name,
 		TTL:     cfc.ResourceRecord.TTL,
 		Proxied: cfc.ResourceRecord.Proxied,
 		Type:    cfc.ResourceRecord.Type,
-		Content: cfc.ResourceRecord.Content,
 	}
+
+	if cfc.ResourceRecord.Type == "SRV" {
+		recordParam.Data = cfc.ResourceRecord.Data
+	} else {
+		recordParam.Content = cfc.ResourceRecord.Content
+	}
+
+	return recordParam, nil
+}
```

Comment:
I'm not sure to see when error can be != than nil in this method. 
Did I missed something ?